crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -42,57 +45,119 @@ impl Ssa {
 impl Context {
     /// Creates a new `Context` object.
     fn new() -> Self {
-        Self { ssa_value_to_acir_var: HashMap::new() }
+        Self { ssa_value_to_acir_var: HashMap::new(), acir_context: AcirContext::new() }
     }
 
     /// Converts SSA into ACIR
-    fn convert_ssa(&mut self, _ssa: Ssa) -> GeneratedAcir {
-        GeneratedAcir {
-            current_witness_index: 0,
-            opcodes: Vec::new(),
-            return_witnesses: Vec::new(),
+    fn convert_ssa(mut self, ssa: Ssa) -> GeneratedAcir {
+        assert_eq!(
+            ssa.functions.len(),
+            1,
+            "expected only a single function to be present with all other functions being inlined."
+        );
+        let main_func = ssa.main();
+        let dfg = &main_func.dfg;
+        let entry_block = &dfg[main_func.entry_block()];
+
+        for param_id in entry_block.parameters() {
+            self.convert_ssa_block_param(*param_id, dfg);
         }
-    }
 
-    /// Converts an SSA value into a `AcirVar`
-    fn convert_ssa_value(
-        &mut self,
-        _value_id: &Id<Value>,
-        _dfg: &DataFlowGraph,
-    ) -> Result<AcirVar, AcirGenError> {
-        todo!()
+        for instruction_id in entry_block.instructions() {
+            self.convert_ssa_instruction(*instruction_id, dfg);
+        }
+
+        self.convert_ssa_return(entry_block.terminator().unwrap(), dfg);
+
+        self.acir_context.finish()
     }
 
-    /// Processes a binary operation and converts the result into an `AcirVar`
-    fn convert_ssa_binary(&mut self, _binary: &Binary, _dfg: &DataFlowGraph) -> AcirVar {
-        todo!()
+    /// Adds and binds an AcirVar for each numeric block parameter
+    ///
+    /// TODO: In the case of references, add AcirVars corresponding to the length of the array
+    /// and bind an ArrayId.
+    fn convert_ssa_block_param(&mut self, param_id: ValueId, dfg: &DataFlowGraph) {
+        let value = dfg[param_id];
+        let param_type = match value {
+            Value::Param { typ, .. } => typ,
+            _ => unreachable!("ICE: Only Param type values should appear in block parameters"),
+        };
+        match param_type {
+            Type::Numeric(..) => {
+                let acir_var = self.acir_context.add_variable();
+                self.ssa_value_to_acir_var.insert(param_id, acir_var);
+            }
+            Type::Reference => {
+                todo!("Alloc array for reference");
```

Comment:
> Alloc array for reference

Please elaborate in the todo macro here. 

I'm okay with having this TODO, though after M1 is complete, all of these TODOs should be cleaned up

Reply:
Added some further detail - is it enough?